x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java

```diff
@@ -151,6 +156,7 @@ public static IndexAction parse(String watchId, String actionId, XContentParser
                 }
             } else if (token == XContentParser.Token.VALUE_STRING) {
                 if (Field.DOC_TYPE.match(currentFieldName, parser.getDeprecationHandler())) {
+                    deprecationLogger.deprecatedAndMaybeLog("watcher_index_action", TYPES_DEPRECATION_MESSAGE);
```

Comment:
It might be nice to add a test for an index action that omits the type parameter completely, as I don't think this was allowed before we introduced the typeless `_bulk` and `index` APIs.